integration_tests/src/main/scala/com/nvidia/spark/rapids/tests/common/BenchUtils.scala

```diff
@@ -549,6 +556,43 @@ class BenchmarkListener(list: ListBuffer[SparkPlanNode]) extends QueryExecutionL
   }
 }
 
+class BenchSparkListener(executionMetrics: ListBuffer[StageMetrics]) extends SparkListener {
+  override def onStageCompleted(stageCompleted: SparkListenerStageCompleted): Unit = {
+    val stageInfo = stageCompleted.stageInfo
+    val taskMetrics = stageInfo.taskMetrics
+
+    val stageMetrics = stageInfo.accumulables.map(acc => Try {
+      val name = acc._2.name.getOrElse("")
```

Comment:
I'm not sure the usage of Try in this case?  What benefit does it add because you have the getORElse - are you expecting _2 to be null?
Also not sure how much it happens but if they don't have names and multiple accumulators are "" we are going to lose that information, perhaps we should use id